Improve TOCO output

benjamintanweihao opened the pull request:
Currently the output is a binary string contain `\n` characters, making the output not very readable. This change makes renders the output as the user expects.

benjamintanweihao committed:
Improve TOCO output

Currently the output is a binary string contain `\n` characters, making the output not very readable.
Also, output streams that have nothing to output show up as `None`.

This change makes renders the output as the user expects.

rmlarsen commented on tensorflow/contrib/lite/python/convert.py:
```diff
@@ -131,8 +131,8 @@ def toco_convert_protos(model_flags_str, toco_flags_str, input_data_str):
     else:
-      raise RuntimeError("TOCO failed see console for info.\n%s\n%s\n" %
-                         (stdout, stderr))
+      raise RuntimeError("TOCO failed. See console for info.\n%s\n%s\n" %
```
Could you add a unit test for this?

benjamintanweihao replied:
Will do! Give me a bit of time to work on it :+1:

benjamintanweihao replied:
@rmlarsen Done!

iganichev committed:
Make defun work under distributed strategies.

The core of the change is have the gradient tape capture
distributed variables instead of plain ResourceVariables.
In other words, we move the distribution awareness from defun
down to tape and rely on distributed variable magic to provide us
with the right variable at runtime.

In tower context, we always watch the container (e.g. MirroredVariable).
In cross tower context, we always watch all the components.

PiperOrigin-RevId: 216430530

tensorflower-gardener committed:
Raises an appropriate error if `add_weight` is called on a Keras network.

PiperOrigin-RevId: 216432358

d0k committed:
[XLA] Allow scatter to share the operand buffer with the output

This avoids a copy.

PiperOrigin-RevId: 216437329

fchollet committed:
Part 2/3 of the update of tf.keras to the Keras 2.2.4 API.

PiperOrigin-RevId: 216442569

jdduke committed:
Internal change

PiperOrigin-RevId: 216442906

Commit:
[TF:XLA] Bump open source abseil revision to 445998d7ac4e5d3c50411d377e3b50e960d2d6c2

PiperOrigin-RevId: 216442983

tensorflower-gardener committed:
Add 'remove' operation to MutableHashTable and MutableDenseHashTable.

PiperOrigin-RevId: 216443201

Commit:
Make lite_test.py run in open source.

PiperOrigin-RevId: 216445964

Commit:
Remove the deprecated created and IS_LOCAL abstractions from activity analysis.

PiperOrigin-RevId: 216446750

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216447412

tensorflower-gardener committed:
[XLA] Added xla::CreateModuleFromProto(...) combining loading module
from proto and verifying it with HloVerifier.

PiperOrigin-RevId: 216447947

d0k committed:
[XLA:GPU] Elide the SequentialThunk when emitting scatter with no copy

We have a 1-element thunk sequence if we're not copying. That's still two
thunks and hlo profiling gets confused if it sees two thunks for the same
instruction and one of them claims to be the whole instruction.

PiperOrigin-RevId: 216448063

tensorflower-gardener committed:
[XLA] Cleanup: Make AllocationTracker::Resolve const.

So that when resolving some global data, we don't have to worry whether
"Resolve" is going to mutate the real data.

PiperOrigin-RevId: 216448145

tensorflower-gardener committed:
Adds an Objective-C API to TensorFlow Lite experimental.

PiperOrigin-RevId: 216451263

Commit:
[XLA:GPU] Use CudnnConvKind in more places.

No functional change.

PiperOrigin-RevId: 216451881

MarkDaoust committed:
Move tflite_convert g3docs, so they will be pulled into the site.

PiperOrigin-RevId: 216452447

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 216452496

tensorflower-gardener committed:
Add support for modeling fast memory close to the processor/gpu

PiperOrigin-RevId: 216453979

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 216455250

miaout17 committed:
Fix lite/kernels:add_test for Clang 8.0.0

PiperOrigin-RevId: 216455772

Commit:
Enable support for lambda functions in static analyses.

The CFG treats lambdas as ordinary expressions. The activity analysis ensures that variables masked by the lambda's arguments are not being tracked.

Note: lambdas do not allow direct modification (we exclude indirect mutation via function or methods).
PiperOrigin-RevId: 216456682

pavithrasv committed:
Update model in keras dist strat learning phase test to return consistent values.

PiperOrigin-RevId: 216461637

smit-hinsu committed:
Automated rollback of commit d78c747e9177fc93d43a580acef2b62eb1420859

PiperOrigin-RevId: 216463443

tensorflower-gardener committed:
Automated rollback of commit 9bd459e4ceba14f9bb1af98d52a109325de952e8

PiperOrigin-RevId: 216463491

mkuperst committed:
[XLA] Add documentation and HLO-level support for multi-value sort.

No support in any of the backends, and not yet exposed through XlaBuilder.

PiperOrigin-RevId: 216465753

tensorflower-gardener committed:
Use Ophints to support TfLite UnidirectionaSequenceLstm and add an e2e test.

Support peephole and num_proj as well.

PiperOrigin-RevId: 216467578

tensorflower-gardener committed:
Add a more verbose error message.

PiperOrigin-RevId: 216471178

miaout17 committed:
Fix lstm_test&layer_norm_lstm_test w/ Clang 8.0.0

PiperOrigin-RevId: 216475683

hsgkim committed:
fixed documentation formatting

gunan committed:
Remove python shebang line from gen_git_source.

PiperOrigin-RevId: 216479972

tensorflower-gardener committed:
Run while loop test that was not being run before.

PiperOrigin-RevId: 216483744

tensorflower-gardener committed:
Delete dead code in batch_scatter_ops_test.

PiperOrigin-RevId: 216483746

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2018-10-10

PiperOrigin-RevId: 216495091

tensorflower-gardener committed:
Emit xla::Or in TensorArrayScatterV3 for PRED types instead of xla::Add

Previosuly we emitted xla::Add what isn't supported by some XLA backend
on PRED types.

PiperOrigin-RevId: 216497939

tensorflower-gardener committed:
Change user_set to an absl::flat_hash_set in HloInstruction.

absl::flat_hash_set have better performance than a std::unordered_set, which can improve overall compile time.
PiperOrigin-RevId: 216498767

tensorflower-gardener committed:
Automated rollback of commit 950cf87104bfee28e2165fe368f66337b8a1336d

PiperOrigin-RevId: 216500702

tensorflower-gardener committed:
Support removing side effecting instructions with RemoveInstructionAndUnusedOperands

If the caller explicitly asks to remove a side effceting instruction
(e.g. all-reduce) then we should respect it instead of silently ignoring
the request.

PiperOrigin-RevId: 216505133

tensorflower-gardener committed:
Support kDomain instructions in the HloMatcher framework

PiperOrigin-RevId: 216525613

Commit:
Use overloaded operators for the assert statement. This should remove the reliance on importing tensorflow in the generated code.

PiperOrigin-RevId: 216528047

tensorflower-gardener committed:
Fix number of outputs when importing tensorflow GraphDef.

Sometimes the actual number of outputs is dictated by one of the attributes of the NodeDef.

PiperOrigin-RevId: 216530696

mrry committed:
[tf.data] `Dataset.make_one_shot_iterator()` inherits the random seed from the calling graph.

This change makes a subtle difference to the behavior of existing
programs that create multiple iterators. Previously, one-shot
iterators would not inherit the graph seed, and so their values would
be non-deterministic (unless explicit seeds were set). After this
change, an iterator will inherit its seed from the outer
graph. Multiple one-shot iterators created from the same dataset will
inherit different seeds, matching the semantics of creating multiple
ops with the same graph seed.

PiperOrigin-RevId: 216532256

mrry committed:
Allow the executor type for a function to be specified as an attr on a function.

This change complements the existing `InstantiateOptions::executor_type`
option, which takes precedence over the attr if both are provided. It
enables the choice of executor to be separated from both the calling
op implementation and the function definition, which simplifies the
use of custom executors in operations that take a function as an attr
(e.g.) `tf.data` and the functional control-flow ops.

PiperOrigin-RevId: 216532778

Commit:
Use lambdas when converting ifexps, since they are now supported.

PiperOrigin-RevId: 216533613

skye committed:
cond_v2: raise an error if pred is a Python bool.

This is to match the existing behavior of tf.cond.

PiperOrigin-RevId: 216534084

ebrevdo committed:
[Grappler] Add RemoveStackStridedSliceSameAxis optimizer.

// Replace operations of the form:
//    x = stack((a_0, a_1, ..., a_{n-1}), axis=k)[:,...,i,...]
// with
//    a_i
// when the strided slice index `i` is applied in the k'th axis.
//
// Similarly, replace operations of the form:
//    x = stack((a_0, a_1, ..., a_{n-1}), axis=k)[:,...,i:i+1,...]
// with
//    expand_dims(a_i, axis=k)
//

PiperOrigin-RevId: 216535346

tensorflower-gardener committed:
remove debug statements

PiperOrigin-RevId: 216536298

jayfurmanek committed:
Make nccl2 bazel configuration platform independent

bbarenblat committed:
Correct a couple of format strings

Change a couple of fscanf-style format strings to use the format macro
constants defined in cinttypes. This quashes -Wformat.

PiperOrigin-RevId: 216545604

tensorflower-gardener committed:
Adds `get_config` and `from_config` to Optimizers V2.

PiperOrigin-RevId: 216546565

aaroey committed:
Add comment about adding --define=with_xla_support=true build option for
mnist_softmax_xla example.

PiperOrigin-RevId: 216547149

Commit:
Port the conditional control flow conversion to the new operators.

PiperOrigin-RevId: 216548561

tensorflower-gardener committed:
Improve shape inference for tf.eye.

PiperOrigin-RevId: 216550243

Commit:
Remove the tensorflow import from generated code.

PiperOrigin-RevId: 216550899

tamaranorman committed:
Remove unused build dependency in gradients_impl on backprop in order to allow reverse dependency.

PiperOrigin-RevId: 216551226

MarkDaoust committed:
Switch to code style, keep link to tensorflow.org

MarkDaoust committed:
Wording.

mkuperst committed:
[XLA] Make sure that the multi-output path of BF16 normalization updates roots.

PiperOrigin-RevId: 216553003

tensorflower-gardener committed:
Call InitMain in tflite_diff_example_test.

PiperOrigin-RevId: 216560608

alextp committed:
Automated rollback of commit 6c40bc717442d56f0b6a60658b05f0549afd69ee.

PiperOrigin-RevId: 216560788

raghuraman-k committed:
Support for shared weights in quantization rewriter.

PiperOrigin-RevId: 216561137

Commit:
Don't decluster ops not supported by TF

There are ops that don't have a TensorFlow kernel and must be run via XLA.
Don't decluster these ops.  If declustering these ops become important in the
future we could perhaps put these on XLA_* devices or put them in a single-node
cluster but YAGNI probably.

PiperOrigin-RevId: 216562037

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216564327

yunxing committed:
Add convenient method result_shape() to hlo module

PiperOrigin-RevId: 216567812

miaout17 committed:
Fix mul_test with Clang 8.0.0

PiperOrigin-RevId: 216570443

tensorflower-gardener committed:
Move from deprecated self.test_session() to self.session() or self.cached_session().

Move to cached_session() if the session is create more than once per test. Move to session() otherwise.

self.test_session() has been deprecated in 9962eb5e84b15e309410071b06c2ed2d6148ed44 as its name confuses readers of the test. Moving to session() instead which slightly changes the semantic of the function:
* the session is not cached anymore (a new session is created).
* the session is closed when exiting the "with" scope.

PiperOrigin-RevId: 216570649

Commit:
Modify getqualifiedname to iterate over a copy of the namespace, to avoid "dictionary changed size during iteration" errors.

PiperOrigin-RevId: 216571661

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216574118

mkuperst committed:
[XLA] Finish NumUniqueInstructionIds() cleanup.

NumUniqueInstructionIds() is no longer a good proxy for the number of instructions. Remove the API, and use the real number of instructions instead of the remaining uses.

PiperOrigin-RevId: 216578465

Commit:
Pass rewriter config into PartitionedCallOp and run the grappler optimization pass

Grappler is disabled by default. Can be controlled through the config argument to enable_eager_execution(), by explicitly setting graph_options.rewrite_options.disable_meta_optimizer to False.

Benchmarks (with grappler enabled):
benchmark_defun_matmul_100_by_784_CPU
  Before: 353.947162628
  After: 322.019815445

benchmark_defun_matmul_2_by_2_CPU
  Before: 76.9423325857
  After: 72.8042999903

benchmark_defun_matmul_2_by_2_CPU_async
  Before: 38.6896689733
  After: 39.6879593531

benchmark_defun_matmul_forward_backward_2_by_2_CPU
  Before: 189.12516435
  After: 182.339771589

benchmark_defun_matmul_forward_backward_2_by_2_CPU_async
  Before: 222.475560506
  After: 245.545005798

benchmark_defun_with_signature
  Before: 120.702934265
  After: 122.84330527

benchmark_defun_with_signature_and_kwargs
  Before: 129.103032748
  After: 140.212361018

benchmark_defun_without_signature
  Before: 63.2988293966
  After: 62.6370668411

benchmark_defun_without_signature_and_with_kwargs
  Before: 73.5487699509
  After: 73.1824000676

PiperOrigin-RevId: 216580246

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 216581784

MarkDaoust committed:
patch the "read the toco guide" button in the index.

PiperOrigin-RevId: 216581868

tensorflower-gardener committed:
Don't use wildcard reshaping when dimensions can be unknown.

PiperOrigin-RevId: 216582137

annarev committed:
Update renames_v2.py file.

PiperOrigin-RevId: 216582173

saxenasaurabh committed:
Support lowering of nested If and While ops.
Removes the LowerIfOpPass and LowerWhileOpPass and adds a unified LowerIfWhilePass.
This change is required because the PRE_PLACEMENT graph optimization pass(https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/graph_execution_state.cc#L531) is run only once and hence can not handle nesting.
Enable the test for nested cond and while in control_flow_ops_py_test.

PiperOrigin-RevId: 216583372

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 216586210

Commit:
Use the conversion options object throughout the process (including during the deferred runtime conversions), and remove its arg_types field.

This is necessary for the correct behavior when converting dynamic functions - converted_call must operate in a manner consistent with the original parameters.

PiperOrigin-RevId: 216587135

Commit:
Don't force random ops (that generate data) to be on the CPU.

It can be slower to generate the data (specially for larger pieces of data) on the host, and then copy to GPU.

Before:
entry {
  name: "MnistEagerGanBenchmark.eager_generate_gpu_batch_64_channels_first"
  iters: 1000
  wall_time: 0.00114285802841
  extras {
    key: "examples_per_sec"
    value {
      double_value: 55999.9566079
    }
  }
}

entry {
  name: "MnistEagerGanBenchmark.eager_generate_gpu_batch_128_channels_first"
  iters: 1000
  wall_time: 0.00138294219971
  extras {
    key: "examples_per_sec"
    value {
      double_value: 92556.2905139
    }
  }
}

entry {
  name: "MnistEagerGanBenchmark.eager_generate_gpu_batch_256_channels_first"
  iters: 1000
  wall_time: 0.00216886901855
  extras {
    key: "examples_per_sec"
    value {
      double_value: 118033.868256
    }
  }
}

After:
entry {
  name: "MnistEagerGanBenchmark.eager_generate_gpu_batch_64_channels_first"
  iters: 1000
  wall_time: 0.000907440900803
  extras {
    key: "examples_per_sec"
    value {
      double_value: 70528.0089793
    }
  }
}

entry {
  name: "MnistEagerGanBenchmark.eager_generate_gpu_batch_128_channels_first"
  iters: 1000
  wall_time: 0.000939697980881
  extras {
    key: "examples_per_sec"
    value {
      double_value: 136213.9779
    }
  }
}

entry {
  name: "MnistEagerGanBenchmark.eager_generate_gpu_batch_256_channels_first"
  iters: 1000
  wall_time: 0.00147917103767
  extras {
    key: "examples_per_sec"
    value {
      double_value: 173069.911105
    }
  }
}

PiperOrigin-RevId: 216587562

yifeif committed:
Update to new toolchain with Clang 8.0.0 r340178.
 - Also update bazel to 0.16.1 because the new toolchain image requires it.

PiperOrigin-RevId: 216590129

ebrevdo committed:
[TF CTC] Add ctc_beam_search_decoder_v2, which disables merge_repeated.

Followup of PRs #15586 and #21826.

PiperOrigin-RevId: 216592105

tensorflower-gardener committed:
create a reusable function for initializing tensorflow from tf lite.

PiperOrigin-RevId: 216597926

Commit:
Automated rollback of commit 6aebb0866718cae2c921e875f3fd74573ee9acc8

PiperOrigin-RevId: 216598193

tensorflower-gardener committed:
Bump open source abseil-cpp revision to f340f773edab951656b19b6f1a77c964a78ec4c2

PiperOrigin-RevId: 216599799

tensorflower-gardener committed:
[XLA] Update TODO references related to all-reduce.

PiperOrigin-RevId: 216600146

alextp committed:
First draft of tf.compat.v1.wrap_function.

PiperOrigin-RevId: 216602214

jdduke committed:
Internal change

PiperOrigin-RevId: 216602362

hawkinsp committed:
[TF:XLA] Don't aggregate writes to TensorArrays that aren't gradient TensorArrays. In the non-XLA case it is an error to write the same entry multiple times.
This saves reading the existing value from memory in cases where it should not be meaningful.

PiperOrigin-RevId: 216604960

jdduke committed:
Only acquire the flex delegate when flex ops present

PiperOrigin-RevId: 216605828

tensorflower-gardener committed:
Don't crash an XRT server if a client leaks a compilation reference.

PiperOrigin-RevId: 216608167

d0k committed:
[XLA] Don't overestimate the amount of bytes a gather reads

PiperOrigin-RevId: 216613367

Commit:
Automated rollback of commit d6a3d6a8295359364c86aecc479e6392bcde0ce4

PiperOrigin-RevId: 216617037

meheffernan committed:
Add HloModule::Clone overload which takes an HloModuleConfig.

PiperOrigin-RevId: 216617647

jsimsa committed:
[tf.data] Changing the background performance modeling to be on by default.

PiperOrigin-RevId: 216618194

rjpower committed:
Turn on worker watchdog in TPUEstimator.

PiperOrigin-RevId: 216618378

Commit:
[XLA:GPU] Rename cudnn convolution passes.

Make them shorter and more consistent.

- CudnnConvolutionFoo -> CudnnConvFoo
- PadInsertion -> CudnnConvPaddingLegalization
- PadForTensorCores -> CudnnConvPadForSpeed (padding channel dimensions from
  3 -> 4 is not a tensor-cores-related optimization and ideally should be run
  on P100s as well).

PiperOrigin-RevId: 216618934

tensorflower-gardener committed:
Allow the XRTCompile op to return the ProgramShape resulted form the XLA compilation.

PiperOrigin-RevId: 216619617

jdduke committed:
Include <cmath> in cost_estimator.h

INFINITY define requires the cmath header.

PiperOrigin-RevId: 216621867

tensorflower-gardener committed:
Adds a simple util to build a GrapplerItem from a MetaGraphDef stored in a file.

PiperOrigin-RevId: 216622520

tensorflower-gardener committed:
Batched per_image_standardization

PiperOrigin-RevId: 216623201

Commit:
[XLA:GPU] Rework CudnnConvPadForSpeed.

 - Make the channel-dims-equal-to-3 heuristic more specific.  The relevant
   cudnn optimization only occurs with input-channels equal to 3 and
   output-channels equal to 32 or 64.

 - Simplify the logic that decided which shapes' dimensions to pad by using a
   clever (if I do say so myself) trick. Really all we need is a new layer of
   indirection which allows us to say "modify the input shape's feature dim"
   and have that be reflected in the lhs shape's feature dim.

 - Rename it to CudnnConvPadForTensorCores.  I initially thought that the
   3-input-channels optimization applied to Pascal, but experimentally this
   seems to be incorrect.

PiperOrigin-RevId: 216623854

rohan100jain committed:
Getting rid of MOVING_AVERAGE_VARIABLES collection usage in quantize_ops.

PiperOrigin-RevId: 216624182

d0k committed:
[XLA:GPU] Allow input fusion into scatter

We fuse everything into the scatter now, and emit two kernels. The first kernel
fills the output buffer with the computation fused into the scatter operand.
The second kernel is a regular scatter, which also contains the fused
operations from the updates and scatter_indices inputs.

PiperOrigin-RevId: 216624225

tensorflower-gardener committed:
Fixes in CompareConstantArrays:
 - Array fields minmax and quantization_params are pointers.
      (so the current check could have false negatives as identical objects
       have different addresses)
 - also compare narrow_range.
      (so the current check could have false positives --- my bad, I added
       narrow_range later and forgot to update this code).

PiperOrigin-RevId: 216626868

kayzhu committed:
[XLA::GPU] Explicitly use {lhs,rhs}_contracting_dimension from DotDimensionNumbers as the reduction dimension for Dot.

PiperOrigin-RevId: 216627100

jsimsa committed:
Automated rollback of commit 9bad98c61f27b60152119bb1c2cfd402c3bf7f3d

PiperOrigin-RevId: 216627219

Commit:
[XLA:GPU] Use HloVerifiedModule in cudnn_conv_pad_for_tensor_cores_test.

No functional change.

PiperOrigin-RevId: 216629980

Commit:
Disable signed-compare warnings in TensorFlow default build.

We have a lot of such warnings and don't make an attempt to clean them up
internally.

PiperOrigin-RevId: 216631010

protoget committed:
Fp16 LSTMBlocKCell and LSTMBlockFusedCell

PiperOrigin-RevId: 216632480

tensorflower-gardener committed:
Automated rollback of commit 3abfe2cd9befa263de57edfae7d4c0d29c9c9182

PiperOrigin-RevId: 216633097

accraze committed:
Update momentum.py

fix docstring line lengths

tensorflower-gardener committed:
Add Range to schema.

PiperOrigin-RevId: 216638239

fchollet committed:
Part 3/4 of the update of tf.keras to the 2.2.4 API.

PiperOrigin-RevId: 216639755

Commit:
Query on Windows as well

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2018-10-11

PiperOrigin-RevId: 216663386

tensorflower-gardener committed:
Only file changed was the tensorcore outfeed manager test, which was passing wrong shape.
Wonder how that could have ever let this CL to go in in the first place.

Automated rollback of commit 905deeaadd41d529461d8a6666e9cf46f0097a8f

PiperOrigin-RevId: 216683052

tensorflower-gardener committed:
Don't use xla::Add on PRED types in tf-xla bridge for TileOp

We are working on disallowing xla::Add on PRED types because they can be
confusing. This is a prerequiste for that to not fail over in the new
shape check.

PiperOrigin-RevId: 216685015

petewarden committed:
Updated model and test data for microcontroller speech example

PiperOrigin-RevId: 216686136

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216686849

akuegel committed:
Support arbitrary many values in KeyValueSort on GPU backend.

PiperOrigin-RevId: 216688700

tensorflower-gardener committed:
Disallow PRED as element type for arithmetic operations in HloVerifier

PiperOrigin-RevId: 216691593

tensorflower-gardener committed:
Automated rollback of commit e5537748753491f803fbddebdcb1cdb710631db9

PiperOrigin-RevId: 216697425

ilya-biryukov committed:
Update downloadable clang to r343880

PiperOrigin-RevId: 216697820

tensorflower-gardener committed:
Automated rollback of commit 9b558126e31d25ec4e82cb4f50033d6eca44349a

PiperOrigin-RevId: 216705354

tensorflower-gardener committed:
Raise the VLOG level for CalculateTensorSize information.

PiperOrigin-RevId: 216705389

Commit:
Account for the possibility of nested lambda functions.

PiperOrigin-RevId: 216706627

Commit:
Have TensorFlow opensource code use tensorflow_estimator module if available.

PiperOrigin-RevId: 216706771

qlzh727 committed:
Update all the RNN tests to also run in eager context.

Also fix the initial_state and constants param in the eager context.

PiperOrigin-RevId: 216712447

tensorflower-gardener committed:
Java support for initializing tensorflow libraries.

PiperOrigin-RevId: 216714487

gunan committed:
Make sure no objects are used before TF module gets initialized.

PiperOrigin-RevId: 216718180

Commit:
Use the functional operator pattern for logical expressions. The current implementation relied on the TF import, which is now gone.

PiperOrigin-RevId: 216719675

tensorflower-gardener committed:
repo.bzl: Increase timeout of repo_ctx.execute

patch command can exceed the 10s timeout limit on some slow machines.

Fixes #22761

PiperOrigin-RevId: 216721485

er91 committed:
Internal change.

PiperOrigin-RevId: 216725062

akuegel committed:
Make sure that all operands and outputs of Sort have the same layout.

Also fix the DotLayout test, it would pass even when commenting out the dot specific logic in GpuLayoutAssignment.

PiperOrigin-RevId: 216726796

Commit:
Modify converted_call to accept a separate argument when the callable is an attribute, representing the attribute's owner.

This gives converted_call a chance to override the attribute lookup, and in turn allows working around the limitation `super` that prevents it from resolving dynamic attributes.

PiperOrigin-RevId: 216727687

asimshankar committed:
[Java]: Release 1.12.0-rc0

PiperOrigin-RevId: 216728781

gunan committed:
Automated rollback of commit 0d054f20851f6156b1af26c35e68f6083bca8e13

PiperOrigin-RevId: 216732038

tensorflower-gardener committed:
Symmetric quantization support for tf.contrib.quantize.

This quantization is configured so that signed integers can be used for inference without biasing, so narrow_range=false implies the negative limit is slightly larger than the positive limit, as is the case for signed two's-complement representation.

PiperOrigin-RevId: 216734571

tensorflower-gardener committed:
Register int64 version of AddN on GPU.

PiperOrigin-RevId: 216736399

meheffernan committed:
Add Unimplemented visitor for token element types to HLO evaluator.
The evaluator would crash when encountering an instruction which produced a
TOKEN shape.

PiperOrigin-RevId: 216737937

yunxing committed:
[Resubmit][XLA] Introduce input/output alias config.

- This CL intruduces input/output alias config in HLO module that allows any HLO pass to configure it. Once the alias_config is set, each backend needs to follow the contract during execution time to make sure the input and output are indeed aliased.

- Copy insertion / buffer assignment and alias analysis has been updated to correctly honor the config and avoid any possible liveness interference.

PiperOrigin-RevId: 216737975

Commit:
Don't segfault if unable to convert non-rectangular inputs to tensor

PiperOrigin-RevId: 216738986

tensorflower-gardener committed:
[XLA] More module proto verification added.

PiperOrigin-RevId: 216745236

tensorflower-gardener committed:
Convert TensorFlow's highwayhash dependency to new third party import method.

PiperOrigin-RevId: 216746022

tensorflower-gardener committed:
Validate the control flow information after subgraph rewrite, since the rewrite may remove some special nodes (e.g. TPUReplicateMetadata) which break the rule.

PiperOrigin-RevId: 216746513

er91 committed:
Move ParseHostComputeCore to side_effect_util.cc. No functionality change.

PiperOrigin-RevId: 216748472

er91 committed:
In control flow functionalization, mark generated If/While with outside compilation attribute if necessary.

PiperOrigin-RevId: 216750679

jsimsa committed:
[tf.data] Performance modeling fixes.

PiperOrigin-RevId: 216753799

gunan committed:
Make sure the ops are not created before TF fully initializes.

PiperOrigin-RevId: 216753969

rxwei committed:
Fix typo

PiperOrigin-RevId: 216754413

mkuperst committed:
[XLA] Expose multi-value sort through the XlaBuilder interface, and add Evaluator support.

Not exposed through the other interfaces yet, e.g. the local computation builder, or the XlaKeyValueSort TF op.

PiperOrigin-RevId: 216754567

Commit:
Make predicted and output types match, so equal op can run.

This will allow https://www.tensorflow.org/tutorials/keras/basic_classification to run in eager mode.

PiperOrigin-RevId: 216754760

tensorflower-gardener committed:
Remove 'not trainable' int types from NCCL gradient tests.

PiperOrigin-RevId: 216755268

Commit:
Calculate batch size correctly when create a Dataset from numpy inputs. Also add a warning message when we don't process all the input samples when we create a Dataset object from input numpy arrays.

PiperOrigin-RevId: 216755953

gunan committed:
Remove GlobalKernelRegistry (returning void*) from the op kernel API.

This was used with RegisterKernels which was removed now.

PiperOrigin-RevId: 216758675

Commit:
[tf.data vectorization] Add vectorizer for "Reshape"

PiperOrigin-RevId: 216759791

hawkinsp committed:
Fix use-of-uninitialized-value bug for bfloat16 in MatrixBandPart.

MatrixBandPart ends up calling bfloat16() to create a zero value, but the implementation of bfloat16::bfloat16() returned an uninitialized value, not zero.

PiperOrigin-RevId: 216761660

gunan committed:
Disable tests that have data dependencies on windows.

Path separators need to be reconfigured on windows before opening files.

PiperOrigin-RevId: 216762475

Commit:
Don't use hardcoded device string when merging device specs.

PiperOrigin-RevId: 216764209

er91 committed:
Add utility function to remove FunctionDef.

PiperOrigin-RevId: 216764308

saeta committed:
[tf.data: NUMA-map-and-batch]: Synchronize access to internal state.

PiperOrigin-RevId: 216770405

fchollet committed:
Update tf.keras version number.

PiperOrigin-RevId: 216770622

rjpower committed:
Fix Keras support in Python 3.

PiperOrigin-RevId: 216770808

Commit:
Add support for tracking deleted variables in activity analysis. Refactor the field names to be more consistent. Remove the unused `returned` field.

PiperOrigin-RevId: 216773791

tensorflower-gardener committed:
Move macros to cc from headers. This will allow us to undef the macros.

PiperOrigin-RevId: 216773822

knightXun committed:
Update kafka_test.sh

allenlavoie committed:
Rely on call op placement for the default device inside a graph function

Device placements inside a function will follow the placement of the call operation unless a device scope is opened inside the function body. This means we no longer need device stacks in defun cache keys, and when functions get serialized in SavedModels the device won't be hard-coded.

Requires adding the current distribution strategy stack to the function cache key, since distribution strategies rely on functions being retraced for each new device (e.g. to access different variables on different devices), and before this CL retracing happened because the function was called with different devices set. This cache key addition does slow things down a bit, but (on my machine at least) the slowdown is more than offset by the gains from not specializing on the device stack.

Baseline before this CL:

entry {
  name: "MicroBenchmarks.benchmark_defun_without_signature"
  iters: 30000
  wall_time: 88.3192300797
  extras {
    key: "examples_per_sec"
    value {
      double_value: 11322.5624714
    }
  }
}

After this CL (includes distribution strategies in cache key):

entry {
  name: "MicroBenchmarks.benchmark_defun_without_signature"
  iters: 30000
  wall_time: 84.1960986455
  extras {
    key: "examples_per_sec"
    value {
      double_value: 11877.0348756
    }
  }
}

Hypothetical world where we didn't have to add distribution strategies to the cache key and also didn't need to add devices (i.e. max speedup to be had by optimizing the distribution strategies cache key addition):

entry {
  name: "MicroBenchmarks.benchmark_defun_without_signature"
  iters: 30000
  wall_time: 72.5416739782
  extras {
    key: "examples_per_sec"
    value {
      double_value: 13785.1795411
    }
  }
}

PiperOrigin-RevId: 216777533

Commit:
Allow the "del" statement in static analysis.

PiperOrigin-RevId: 216783344

mkuperst committed:
[XLA] Add key-value sort test with int64 values.

Also some small improvements to HloSortInstruction.

PiperOrigin-RevId: 216784851

Commit:
Implement lazy compilation for XLA

This CL implements lazy compilation for XLA with a very simple policy: compile a
cluster for a specific signature the second time we see the signature.  This
policy is only for bootstrapping; I suspect we will have to substantially tune
this based on real world workloads.

This CL is organized as follows:

 - Mark the compilation_successful output / input in the _XlaCompile/_XlaRun
   kernels as HostMem.  This was a bug in the CL that introduced
   _XlaCompile/_XlaRun, but was unnoticed till now because we never read the
   compilation_successful boolean.
    * Also mark the key output / input as HostMem for symmetry even though
     DT_STRING tensors are always placed on the host.

 - Add a couple of new TF node matchers to make it easier to unit-test the
   updated build_xla_ops pass.

 - Update some tests that relied on JIT compilation to have a "warmup" phase so
   that compilation is guaranteed.

 - The main part of the CL is in kernels/xla_ops and jit/xla_compilation_cache.

PiperOrigin-RevId: 216786799

annarev committed:
Traverse through all enum names even if they have duplicate values.

PiperOrigin-RevId: 216787178

annarev committed:
Removing deprecated endpoints in V2. Also, deprecating "saved_model.loader.load" which is replaced by "saved_model.load" and saved_model.main_op, which will be replaced by "saved_model.main_op" in V2.

PiperOrigin-RevId: 216788275

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216788277

skye committed:
Correctly check for while loops in AutomaticControlDependencies.

AutomaticControlDependencies isn't implemented for v1 while_loops, but
didn't check parent control flow contexts for WhileLoopContexts.

PiperOrigin-RevId: 216788530

mrry committed:
[tf.data] Add an experimental dataset that introduces a sleep into the pipeline.

This dataset will enable us to experiment with "delay scheduling", which could lessen the interference between input pipeline work and critical path activities (such as launching GPU kernels, etc.).

PiperOrigin-RevId: 216788581

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 216791162

tayo committed:
Changes NMS XLA implementation to match that of CPU.

PiperOrigin-RevId: 216796208

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216796407

fdxmw committed:
TPUEstimator: Initialize dataset iterators in parallel.

PiperOrigin-RevId: 216796462

kayzhu committed:
[XLA:GPU] Adding a test case for Scatter where GPU implementation fails.

PiperOrigin-RevId: 216798034

protoget committed:
LSTMBlockCell Fp16 python change

PiperOrigin-RevId: 216798109

Commit:
Removed TODOs that are obsolete.

PiperOrigin-RevId: 216798710

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 216799716

annarev committed:
Sort members when adding enum names in TF API traverser.

PiperOrigin-RevId: 216809329

tensorflower-gardener committed:
Batched per_image_standardization

PiperOrigin-RevId: 216813273

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216815486

tensorflower-gardener committed:
Fix typo in documentation.

Eps is defined as '1 + eps != 1', not as '1 + eps != eps'.

PiperOrigin-RevId: 216824143

benjamintanweihao commented:
Yup this is completely my bad. I’ll fix it. 

> On 12 Oct 2018, at 4:30 PM, Jason Zaman <notifications@github.com> wrote:
> 
> @benjamintanweihao hey, um, something went wrong with this. Did you merge master into your topic branch? It is trying to merge 300 commits in which is clearly wrong. Can you rebase your branch on origin/master and make sure it has no merge commits and its only one commit off master? then force push the branch to your repo and it should fix things.
> 
> something like:
> 
> git fetch --all
> git checkout patch-2
> git rebase origin/master
> git log origin/master..patch-2 # verify its right
> git push <your repo> patch-2 --force
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2018-10-12

PiperOrigin-RevId: 216829748

er91 committed:
Add a utility function to build node name to node index.

PiperOrigin-RevId: 216853788

tensorflower-gardener committed:
Add RaggedTensors to tf.core. Moving the RaggedRange op kernel.

PiperOrigin-RevId: 216865682

tensorflower-gardener committed:
Move from deprecated self.test_session() to self.session() or self.cached_session().

Move to cached_session() if the session is create more than once per test. Move to session() otherwise.

self.test_session() has been deprecated in 9962eb5e84b15e309410071b06c2ed2d6148ed44 as its name confuses readers of the test. Moving to session() instead which slightly changes the semantic of the function:
* the session is not cached anymore (a new session is created).
* the session is closed when exiting the "with" scope.

PiperOrigin-RevId: 216868101

tensorflower-gardener committed:
Move from deprecated self.test_session() to self.session() or self.cached_session().

Move to cached_session() if the session is create more than once per test. Move to session() otherwise.

self.test_session() has been deprecated in 9962eb5e84b15e309410071b06c2ed2d6148ed44 as its name confuses readers of the test. Moving to session() instead which slightly changes the semantic of the function:
* the session is not cached anymore (a new session is created).
* the session is closed when exiting the "with" scope.

PiperOrigin-RevId: 216868175

eddie-zhou committed:
Make SDCA use VariableV1 (locking defaults change).

PiperOrigin-RevId: 216868252

meheffernan committed:
Add methods to xla::Compiler which operate on an HloModuleGroup.
Methods which currently take a std::vector<HloModule> now take a HloModuleGroup.

PiperOrigin-RevId: 216869028

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 216872436

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 216876525

gunan committed:
Auto load kernel libraries under tensorflow/core/kernels when TF kernel registry is accessed.

PiperOrigin-RevId: 216878902

meheffernan committed:
Automated rollback of commit 51f0eb5849be0f9ce20e5eb8370158088711f19d

PiperOrigin-RevId: 216879444

wangtz committed:
FeatureLayer: Nit: Slightly speed it up by pre-sorting the columns by name.

PiperOrigin-RevId: 216880147

tensorflower-gardener committed:
Call InitTensorflow in benchmark_main.cc

PiperOrigin-RevId: 216881319

tensorflower-gardener committed:
Add timeout mechanism for Grappler optimizers. Sprinkle calls to GRAPPLER_RETURN_IF_DEADLINE_EXCEEDED() in the most expensive optimizers.

PiperOrigin-RevId: 216883291

tensorflower-gardener committed:
Bugfix for optimized TransposeConv(). Was incorrectly interpreting weights as HWIO, instead of HWOI.

PiperOrigin-RevId: 216883371

tensorflower-gardener committed:
Handle strings in tooling_util.cc

PiperOrigin-RevId: 216886885

lamberta committed:
UPdate tflite converter diagram.
Rename URL paths.

PiperOrigin-RevId: 216889190

mrry committed:
Fix a piece of lint in the docstring for `tf_export` (non-matching quote characters).

PiperOrigin-RevId: 216890307

mrry committed:
[tf.data] Export `tf.data.experimental.AUTOTUNE`.

PiperOrigin-RevId: 216890862

Commit:
Simplify implementation of shared weights support to not depend on string names.

Instead we just ensure that all consumers of a target tensor use the inserted FakeQuant tensor instead.

PiperOrigin-RevId: 216891274

tensorflower-gardener committed:
Make XRT handles random 64-bit integers instead of increasing dense integers. This somewhat reduces the probability of a client accidentally using a stale handle without realizing it.

PiperOrigin-RevId: 216894405

Commit:
Copy GPU tensors to host before sending remotely.

AsProtoTensorContent doesn't work if the tensor buffer is on the GPU.

PiperOrigin-RevId: 216895001

miaout17 committed:
Support real custom ops for Toco --allow_eager_ops flow.

PiperOrigin-RevId: 216896261

skye committed:
Have cond_v2 and while_v2 use LazyLoader to resolve circular dependencies.

PiperOrigin-RevId: 216896672

gmagogsfm committed:
Fix a typo in xla.estimator_model_fn docstring

PiperOrigin-RevId: 216898670

gunan committed:
Automated rollback of commit 7294f306c0e144db5340b713c903c45e79105e5e

PiperOrigin-RevId: 216898976

Commit:
[tf.data vectorization] Handle negative "axis" attr values for Unpack vectorizer

PiperOrigin-RevId: 216898989

Commit:
Internal change.

PiperOrigin-RevId: 216899775

tensorflower-gardener committed:
Automated rollback of commit 14f86407043fec62890fa00c246f5968b12a15f1

PiperOrigin-RevId: 216903862

d0k committed:
[XLA:GPU] Adapt test for cudnn conv picker pass rename

This now fails in some configurations and is obviously doing the wrong thing.

PiperOrigin-RevId: 216907441

ebrevdo committed:
Fix shape inference segfault in ScatterNdUpdateShape.

Fixes #22013.

PiperOrigin-RevId: 216909792

ebrevdo committed:
[Grappler] Enable RemoveStackStridedSliceSameAxis optimizer.

PiperOrigin-RevId: 216912523

meheffernan committed:
Rollforward with build fix to MLIR TPU compiler.
Also renamed some methods to avoid "hides overloaded virtual function" compilation error which only appears in the "Builder" analysis in Critique. See b/11765370.

*** Original change description ***

Automated rollback of commit 51f0eb5849be0f9ce20e5eb8370158088711f19d

PiperOrigin-RevId: 216914046

tensorflower-gardener committed:
Speed up Grappler by using FlatMap/FlatSet instead of std::unordered_map/std::unordered_set in NodeMap and SetVector.
Fix a latent bug in DedupComputations where we depended on pointer stability in NodeMap.

Benchmark numbers for running Grappler on the Transformer model:

name                                old time/op             new time/op             delta
BM_OptimizeTransformer              6.55s ? 0%              5.65s ? 0%   ~     (p=1.000 n=1+1)

name                                old allocs/op           new allocs/op           delta
BM_OptimizeTransformer              24.4M ? 0%              23.3M ? 0%   ~     (p=1.000 n=1+1)

name                                old peak-mem(Bytes)/op  new peak-mem(Bytes)/op  delta
BM_OptimizeTransformer               123M ? 0%               123M ? 0%   ~     (p=1.000 n=1+1)

PiperOrigin-RevId: 216918669

mkuperst committed:
[XLA] Allow whitespace in shapes and layouts in HLO text.

PiperOrigin-RevId: 216919890

k-w-w committed:
Replacing legacy_init_op argument in SavedModelBuilder with main_op.

PiperOrigin-RevId: 216920955

d0k committed:
[XLA:GPU] Fix scatter oob check computation

This was comparing the index after adding it to the window, and then comparing
against the window dimension. This means that the bounds check was only correct
for the first element of a window. Instead compare the scatter index, which is
the same for all elements of a window.

PiperOrigin-RevId: 216921512

protoget committed:
Fix a tiny bug

PiperOrigin-RevId: 216923314

peterma123456789 committed:
Slight change to the API of VirtualScheduler; Include RunMetadata in AnalyticalCostEstimator; Add more fields in CostGraphDef in AnalyticalCostEstimator

PiperOrigin-RevId: 216924665

miaout17 committed:
Blacklist `HashTableV2` for Flex in Toco converter.

PiperOrigin-RevId: 216925873

Commit:
Register Assert op for GPU.

PiperOrigin-RevId: 216925919

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216926812

raghuraman-k committed:
Enable quantization of activations after add and mul operations in a graph. This change allows for full quantization of mobilenet ssd models.

PiperOrigin-RevId: 216928860

tensorflower-gardener committed:
Automated rollback of commit 0608a3f0dbbac03aeb84966d2ea9a24a694ebbce

PiperOrigin-RevId: 216931739

gunan committed:
Make AWS, GCP, HDFS, Kafka and Ignite default on, but introduce the ability to
disable them

Fixes #22819

PiperOrigin-RevId: 216931905

tensorflower-gardener committed:
Performance improvement for distillation of DNNs to trees.
* Move the DNN network to the training-only graph so it is not in the eval if we're doing distillation. This makes the inference much faster because we no longer run the DNN.
* Simplify use_core_version logic. Now we always return model_fn for contrib and estimator_spec for core.

PiperOrigin-RevId: 216932818

tensorflower-gardener committed:
Add some benchmarks for parallal_for.

PiperOrigin-RevId: 216932970

Commit:
[TF:XLA] Bump open source abseil revision to a00bdd176d66ef0b417d9576052a19091fbdf891

PiperOrigin-RevId: 216934705

qlzh727 committed:
Update Keras RNN to support nested input/output/states.

PiperOrigin-RevId: 216934951

tensorflower-gardener committed:
This CL switches the custom-registered-gradient of nested Functions from using the C++ SymGrad to the correct forward & backward passes. This allows gradients of defuns nested > 2 levels deep to work correctly.

This CL also adds a number of unit tests to make sure various nestings of gradient & defuns work. (Note: unit tests that combine multiple levels of gradient tapes & multiple levels of defun still need to be added).

PiperOrigin-RevId: 216939066

Commit:
Expose QuantizeAndDequantizeV2 to open-source Python API.

PiperOrigin-RevId: 216939077

gunan committed:
Remove tensorflow_no_py_deps from configure script.

This config breaks out of the box usage of "bazel test" and hurts development.

PiperOrigin-RevId: 216939335

Commit:
Explicitly set jdk8 in ci_parameterized_build.sh

PiperOrigin-RevId: 216946217

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216947070

jdduke committed:
Internal change

PiperOrigin-RevId: 216947172

tensorflower-gardener committed:
Open source mid-level TPU embedding API.

PiperOrigin-RevId: 216947827

aselle committed:
Fix TensorFlow Lite Convert on Windows

NamedTemporaryFile's need to be used differently because Windows
cannot have multiple open handles to a single file.

PiperOrigin-RevId: 216948856

tensorflower-gardener committed:
Switches defun gradients to output None for unconnected gradients instead of 0.0.

PiperOrigin-RevId: 216949450

kayzhu committed:
[XLA]Enable more DataTypes for running on the XLA:INTERPRETER backend.

PiperOrigin-RevId: 216949811

jdduke committed:
Automated rollback of commit 1a77f1876e8138c84f96e639e464096294e9f389

PiperOrigin-RevId: 216951012

tensorflower-gardener committed:
Export the host program shape (with layouts) in the HLO module proto.

PiperOrigin-RevId: 216951291

tensorflower-gardener committed:
Extend the PrintV2 op to write tensors to file.
  If the output_stream attribute is started with "file://", the string that follows will be interpreted as a file path. Then PrintV2 will write the tensors to that file.

PiperOrigin-RevId: 216952073

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 216953764

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 216956589

tensorflowbutler commented:
Nagging Assignee @rmlarsen: It has been 19 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

benjamintanweihao commented:
Will do it tomorrow. :)

> On 27 Nov 2018, at 10:30 PM, Jason Zaman <notifications@github.com> wrote:
> 
> @benjamintanweihao hey, TFLite moved out of contrib, can you update this commit again please?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

rmlarsen commented:
@benjamintanweihao are you planning to move forward with this?

rmlarsen commented:
Thanks. Please resolve the conflicts.

benjamintanweihao commented:
Argh. I will redo this.
